test/helpers/shouldFail.js

```diff
@@ -0,0 +1,31 @@
+const should = require('chai')
+  .should();
+
+async function shouldFailWithMessage (promise, message) {
+  try {
+    await promise;
+  } catch (error) {
+    error.message.should.include(message, 'Wrong failure type');
+    return;
+  }
+
+  should.fail(`Expected '${message} failure not received`);
+}
+
+function reverting (promise) {
```

Comment:
Either add `async` here, or remove it in the next two functions for consistency. I'd add it here.

Reply:
I forgot to remove it from the other two ones. It makes no sense to mark it `async` unless we `await` inside it, and since all `await` in this case is return a promise, I'd rather just do it directly.

Reply:
I'd like you to reconsider making the functions `async`. It makes the return value of the function always be a promise regardless of the type of the argument. It's also like having a type annotation, and likewise it helps when reading the code IMO.